pyproject.toml

```diff
@@ -31,6 +31,7 @@ dependencies = [
   "requests",
   "semver",
   "tomli",
+  "tomlkit",
```

Comment:
nitpick: Care enough to add an extra commit switching our own logic from tomli to tomlkit :)? It should be really simple since there are only 2 files where it's imported: `pip.py` and `nox.py`. Note we won't be able to get rid of it completely, it'll still remain in the requirements files as it's pulled in by something else, but we could replace it in `pyproject.toml`.